src/traits.rs

```diff
@@ -848,11 +878,491 @@ pub trait PrimInt
+       self & (cast::<_,Self>(0).unwrap() - self)
+   }
+
+   /// Set the least significant zero bit of `self` to 1 and all of the
+   /// rest to 0.
+   ///
+   /// # Examples
+   ///
+   /// ```
+   /// use num::traits::PrimInt;
+   ///
+   /// let n = 0b0101;
+   /// let s = 0b0010;  
+   ///
+   /// assert_eq!(n.isolate_least_significant_zero(), s);
+   /// ```
+   fn isolate_least_significant_zero(self) -> Self {
+       (!self) & (self + cast(1).unwrap())
+   }
+
+   /// Reset the trailing 1's in `self`.
+   ///
+   /// # Examples
+   ///
+   /// ```
+   /// use num::traits::PrimInt;
+   ///
+   /// let n = 0b0110_1111;
+   /// let s = 0b0110_0000;  
+   ///
+   /// assert_eq!(n.reset_trailing_ones(), s);
+   /// ```
+   fn reset_trailing_ones(self) -> Self {
+       self & (self + cast(1).unwrap())
+   }
+
+   /// Set all of the trailing 0's in `self`.
+   ///
+   /// # Examples
+   ///
+   /// ```
+   /// use num::traits::PrimInt;
+   ///
+   /// let n = 0b0110_0000u8;
+   /// let s = 0b0111_1111u8;  
+   ///
+   /// assert_eq!(n.set_trailing_zeros(), s);
+   /// ```
+   fn set_trailing_zeros(self) -> Self {
+       self | (self - cast(1).unwrap())
+   }
+
+   /// Returns a mask with all of the trailing 0's of `self` set.
+   ///
+   /// # Examples
+   ///
+   /// ```
+   /// use num::traits::PrimInt;
+   ///
+   /// let n = 0b0110_0000u8;
+   /// let s = 0b0001_1111u8;  
+   ///
+   /// assert_eq!(n.mask_trailing_zeros(), s);
+   /// ```
+   fn mask_trailing_zeros(self) -> Self {
+       (!self) & (self - cast(1).unwrap())
+   }
+
+   /// Returns a mask with all of the trailing 1's of `self` set.
+   ///
+   /// # Examples
+   ///
+   /// ```
+   /// use num::traits::PrimInt;
+   ///
+   /// let n = 0b0101_1111u8;
+   /// let s = 0b0001_1111u8;  
+   ///
+   /// assert_eq!(n.mask_trailing_ones(), s);
+   /// ```
+   fn mask_trailing_ones(self) -> Self {
+       !((!self) | (self + cast(1).unwrap()))
+   }
+   
+   /// Returns a mask with all of the trailing 0's of `self` set and the least
+   /// significant 1 bit set.
+   ///
+   /// # Examples
+   ///
+   /// ```
+   /// use num::traits::PrimInt;
+   ///
+   /// let n = 0b0101_0000u8;
+   /// let s = 0b0001_1111u8;  
+   ///
+   /// assert_eq!(n.mask_trailing_zeros_and_least_significant_one(), s);
+   /// ```
+   fn mask_trailing_zeros_and_least_significant_one(self) -> Self {
+       (self - cast(1).unwrap()) ^ self
+   }
+
+   /// Returns a mask with all of the trailing 1's of `self` set and the least
+   /// significant 0 bit set.
+   ///
+   /// # Examples
+   ///
+   /// ```
+   /// use num::traits::PrimInt;
+   ///
+   /// let n = 0b0101_1111u8;
+   /// let s = 0b0011_1111u8;  
+   ///
+   /// assert_eq!(n.mask_trailing_ones_and_least_significant_zero(), s);
+   /// ```
+   fn mask_trailing_ones_and_least_significant_zero(self) -> Self {
+       self ^ (self + cast(1).unwrap())
+   }
+
+   fn general_reverse_bits(self, subword_bits: u32, group_subwords: u32) -> Self;
```

Comment:
Since this is undocumented, I guess it's meant to be an implementation detail, but all functions in a pub trait are exposed.  You'll need to pull this out and implement it separately.